### ISSUE TITLE
mixed precision error casting

### ISSUE THREAD

**Issue by Shiro-LK** [faustomorales/vit-keras#11]
Hello,
First of all, thank you for this nice re-implementation in keras with the pretrained weights available.

It is just to raise an issue when using mixed precision , it seems to be related to some casting. 

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/opt/conda/lib/python3.7/site-packages/tensorflow/python/framework/op_def_library.py in _apply_op_helper(op_type_name, name, **keywords)
    437               preferred_dtype=default_dtype,
--> 438               as_ref=input_arg.is_ref)
    439           if input_arg.number_attr and len(

/opt/conda/lib/python3.7/site-packages/tensorflow/python/framework/ops.py in internal_convert_n_to_tensor(values, dtype, name, as_ref, preferred_dtype, ctx)
   1607             preferred_dtype=preferred_dtype,
-> 1608             ctx=ctx))
   1609   return ret

/opt/conda/lib/python3.7/site-packages/tensorflow/python/profiler/trace.py in wrapped(*args, **kwargs)
    162           return func(*args, **kwargs)
--> 163       return func(*args, **kwargs)
    164 

/opt/conda/lib/python3.7/site-packages/tensorflow/python/framework/ops.py in convert_to_tensor(value, dtype, name, as_ref, preferred_dtype, dtype_hint, ctx, accepted_result_types)
   1508           "Tensor conversion requested dtype %s for Tensor with dtype %s: %r" %
-> 1509           (dtype.name, value.dtype.name, value))
   1510     return value

ValueError: Tensor conversion requested dtype float32 for Tensor with dtype bfloat16: <tf.Tensor 'Placeholder:0' shape=(None, 576, 768) dtype=bfloat16>

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-41-d11fa3fc8440> in <module>
     35         K.clear_session()
     36         with strategy.scope():
---> 37             model = compile_new_model_vit(cfg, cfg['name'][num_model], with_sgd=False)
     38 
     39 

<ipython-input-38-6c20ae4b4197> in compile_new_model_vit(cfg, name, with_sgd)
     13 def compile_new_model_vit(cfg, name, with_sgd=False):
     14     with strategy.scope():
---> 15         model = get_model_vit(cfg, name)
     16 
     17         losses = CategoricalCrossentropyCustom(label_smoothing = cfg['smoothing'])

<ipython-input-38-6c20ae4b4197> in get_model_vit(cfg, name)
      2     model_input = tf.keras.Input(shape=(cfg['resize'][0], cfg['resize'][1], 3), name='inputs')
      3     image_size = 384
----> 4     x = vit.vit_b16(image_size=image_size,activation='sigmoid', pretrained=True, include_top=False, pretrained_top=False)(model_input)
      5     #x1 = tf.keras.layers.GlobalAveragePooling2D(name='avg_pool')(x)
      6     #x2 = tf.keras.layers.GlobalMaxPooling2D(name='max_pool')(x)

/opt/conda/lib/python3.7/site-packages/vit_keras/vit.py in vit_b16(image_size, classes, activation, include_top, pretrained, pretrained_top, weights)
    146         activation=activation,
    147         include_top=include_top,
--> 148         representation_size=768 if weights == "imagenet21k" else None,
    149     )
    150     if pretrained:

/opt/conda/lib/python3.7/site-packages/vit_keras/vit.py in build_model(image_size, patch_size, num_layers, hidden_size, num_heads, name, mlp_dim, classes, dropout, activation, include_top, representation_size)
     75     )(x)
     76     y = tf.keras.layers.Reshape((y.shape[1]*y.shape[2], hidden_size))(y)
---> 77     y = layers.ClassToken(name="class_token")(y)
     78     y = layers.AddPositionEmbs(name="Transformer/posembed_input")(y)
     79     for n in range(num_layers):

/opt/conda/lib/python3.7/site-packages/tensorflow/python/keras/engine/base_layer.py in __call__(self, *args, **kwargs)
    950     if _in_functional_construction_mode(self, inputs, args, kwargs, input_list):
    951       return self._functional_construction_call(inputs, args, kwargs,
--> 952                                                 input_list)
    953 
    954     # Maintains info about the `Layer.call` stack.

/opt/conda/lib/python3.7/site-packages/tensorflow/python/keras/engine/base_layer.py in _functional_construction_call(self, inputs, args, kwargs, input_list)
   1089         # Check input assumptions set after layer building, e.g. input shape.
   1090         outputs = self._keras_tensor_symbolic_call(
-> 1091             inputs, input_masks, args, kwargs)
   1092 
   1093         if outputs is None:

/opt/conda/lib/python3.7/site-packages/tensorflow/python/keras/engine/base_layer.py in _keras_tensor_symbolic_call(self, inputs, input_masks, args, kwargs)
    820       return nest.map_structure(keras_tensor.KerasTensor, output_signature)
    821     else:
--> 822       return self._infer_output_signature(inputs, args, kwargs, input_masks)
    823 
    824   def _infer_output_signature(self, inputs, args, kwargs, input_masks):

/opt/conda/lib/python3.7/site-packages/tensorflow/python/keras/engine/base_layer.py in _infer_output_signature(self, inputs, args, kwargs, input_masks)
    861           # TODO(kaftan): do we maybe_build here, or have we already done it?
    862           self._maybe_build(inputs)
--> 863           outputs = call_fn(inputs, *args, **kwargs)
    864 
    865         self._handle_activity_regularization(inputs, outputs)

/opt/conda/lib/python3.7/site-packages/vit_keras/layers.py in call(self, inputs)
     20         batch_size = tf.shape(inputs)[0]
     21         cls_broadcasted = tf.broadcast_to(self.cls, [batch_size, 1, self.hidden_size])
---> 22         return tf.concat([cls_broadcasted, inputs], 1)
     23 
     24 

/opt/conda/lib/python3.7/site-packages/tensorflow/python/util/dispatch.py in wrapper(*args, **kwargs)
    199     """Call target, and fall back on dispatchers if there is a TypeError."""
    200     try:
--> 201       return target(*args, **kwargs)
    202     except (TypeError, ValueError):
    203       # Note: convert_to_eager_tensor currently raises a ValueError, not a

/opt/conda/lib/python3.7/site-packages/tensorflow/python/ops/array_ops.py in concat(values, axis, name)
   1675           dtype=dtypes.int32).get_shape().assert_has_rank(0)
   1676       return identity(values[0], name=name)
-> 1677   return gen_array_ops.concat_v2(values=values, axis=axis, name=name)
   1678 
   1679 

/opt/conda/lib/python3.7/site-packages/tensorflow/python/ops/gen_array_ops.py in concat_v2(values, axis, name)
   1206   _attr_N = len(values)
   1207   _, _, _op, _outputs = _op_def_library._apply_op_helper(
-> 1208         "ConcatV2", values=values, axis=axis, name=name)
   1209   _result = _outputs[:]
   1210   if _execute.must_record_gradient():

/opt/conda/lib/python3.7/site-packages/tensorflow/python/framework/op_def_library.py in _apply_op_helper(op_type_name, name, **keywords)
    464                               (prefix, dtype.name))
    465             else:
--> 466               raise TypeError("%s that don't all match." % prefix)
    467           else:
    468             raise TypeError(

TypeError: Tensors in list passed to 'values' of 'ConcatV2' Op have types [float32, bfloat16] that don't all match.


```

**Comment by Shiro-LK** [comment]
issue solved.
I had a pull request to make it officially if needed.

**Comment by faustomorales** [comment]
Thanks so much! Your fix is now included in [v0.0.11](https://pypi.org/project/vit-keras/0.0.11/).
